ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -12,204 +12,142 @@
+    return Array.isArray(parsedCoords) && Array.isArray(parsedCoords[0])
+  } catch (e) {
+    return false
   }
-  isValidPolygon(coordinates) {
-    if (!this.is2DArray(coordinates)) {
-      this.invalidMessage = 'Not an acceptable value.'
-      return false
+}
+
+const BaseLine = props => {
+  const { label, geometryKey, setState, unitKey, widthKey, mode } = props
+  const [currentValue, setCurrentValue] = useState(
+    JSON.stringify(props[geometryKey])
+  )
+  const [baseLineError, setBaseLineError] = useState(initialErrorState)
+  const [bufferError, setBufferError] = useState(initialErrorState)
+
+  useEffect(
+    () => {
+      const { geometryKey } = props
+      setCurrentValue(JSON.stringify(props[geometryKey]))
+    },
+    [props.polygon, props.line]
+  )
+
+  function testValidity() {
+    if (!is2DArray(currentValue)) {
+      return { error: true, message: 'Not an acceptable value' }
     }
     try {
-      this.validateListOfPoints(JSON.parse(coordinates))
-      return true
+      return validateListOfPoints(JSON.parse(currentValue), mode)
     } catch (e) {
-      return false
+      //do nothing
     }
   }
-  convertWkt(value, numCoords) {
-    const coordinatePairs = value.match(coordinatePairRegex)
-    if (!coordinatePairs || coordinatePairs.length < numCoords) {
-      return value
-    }
-    const coordinates = coordinatePairs.map(coord => coord.replace(' ', ','))
-    return `[[${coordinates.join('],[')}]]`
-  }
-  convertMultiWkt(isPolygon, value) {
-    if (isPolygon && !value.endsWith(')))')) {
-      return value
-    } else if (!value.endsWith('))')) {
-      return value
-    }
-    const splitter = isPolygon ? '))' : ')'
-    const numPoints = isPolygon ? 4 : 2
-    let shapes = value
-      .split(splitter)
-      .map(shape => shape.match(coordinatePairRegex))
-    shapes = shapes
-      .filter(shape => shape !== null && shape.length >= numPoints)
-      .map(shape =>
-        shape.map(coordinatePair => coordinatePair.replace(' ', ','))
-      )
-    return shapes.length === 0
-      ? value
-      : shapes.length === 1
-        ? `[[${shapes[0].join('],[')}]]`
-        : `[${shapes.map(shapeCoords => `[[${shapeCoords.join('],[')}]]`)}]`
-  }
+
+  return (
+    <div>
+      <div className="input-location">
+        <TextField
+          label={label}
+          value={currentValue}
+          onChange={value => {
+            value = convertWktString(value.trim())
+            setCurrentValue(value)
+            try {
+              //handle case where user clears input; JSON.parse('') would throw an error and maintain previous state
+              if (value === '') {
+                setState(geometryKey, undefined)
+              } else {
+                setState(geometryKey, JSON.parse(value))
+              }
+            } catch (e) {
+              // do nothing
+            }
+          }}
+          onBlur={() => setBaseLineError(testValidity())}
```

Comment:
The `testValidity` logic could be moved to a `validateLineAndPolygon` function in validation.js and this call could be replaced with `validateGeo(mode, currentValue)`